src/network/netlink.rs

```diff
@@ -212,16 +212,22 @@ impl Socket {
         Ok(())
     }
 
-    fn create_route_msg(route: &Route) -> RouteMessage {
+    fn create_route_msg(route: &Route, metric: Option<u32>) -> RouteMessage {
         let mut msg = RouteMessage::default();
 
+        let final_metric = metric.unwrap_or(100);
+
         msg.header.table = RT_TABLE_MAIN;
         msg.header.protocol = RTPROT_STATIC;
         msg.header.scope = RT_SCOPE_UNIVERSE;
         msg.header.kind = RTN_UNICAST;
 
         let (dest_vec, dest_prefix, gateway_vec) = match route {
             Route::Ipv4 { dest, gw } => {
+                info!(
+                    "Adding route to {} via {} with metric {}",
+                    dest, gw, final_metric
+                );
```

Comment:
This is inconsistent with the other calls which do not log. In principle I am ok with this but we should be consistent across the code base. Currently logs are all over the place and it is not really clear what should be trace, debug and info level? 

Also you could just `"{}", route` since the Display trait is implemented which should allow you do not duplicate the log line twice.